packages/transit-vehicle-overlay/src/styled.ts

```diff
@@ -0,0 +1,136 @@
+import coreUtils from "@opentripplanner/core-utils";
+import { FC, HTMLAttributes } from "react";
+import styled, { css } from "styled-components";
+import { TransitVehicle } from "@opentripplanner/types";
+
+// Need this to find null or undefined values while not including zero which is a valid value.
+import { isNull } from "./utils/strings";
+
+export interface IconContainerProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * The padding around icons, in pixels.
+   */
+  padding?: number;
+  /**
+   * The size of the icon in pixels.
+   */
+  pixels?: number;
+  /**
+   * The transit vehicle for which to render a symbol.
+   */
+  vehicle: TransitVehicle;
+}
+
+interface ColorProps {
+  backgroundColor: string;
+  foregroundColor: string;
+}
+
+// CSS helper functions.
+const getPixels = props => (isNull(props.pixels) ? 15 : props.pixels);
+const getPadding = props => (isNull(props.padding) ? 5 : props.padding);
+const getForegroundColor = props => props.foregroundColor;
+
+/**
+ * Computes color props to simplify the CSS filler code.
+ */
+function getColorProps(defaultColor: string) {
+  return (props: IconContainerProps): ColorProps => {
+    const routeColor = props.vehicle.routeColor || defaultColor;
+    return {
+      backgroundColor: routeColor,
+      foregroundColor: coreUtils.route.getMostReadableTextColor(routeColor)
+    };
+  };
+}
+
+/**
+ * Displays a circle with basic settings.
+ */
+export const Circle = styled.div<IconContainerProps>`
+  background: #eee;
+  border: 2px solid #333;
+  border-radius: ${getPixels}px;
+  cursor: default;
+  height: ${getPixels}px;
+  line-height: ${getPixels}px;
+  padding: ${getPadding}px;
+  position: relative;
+  text-align: center;
+  transition: all 0.1s ease-in-out;
```

Comment:
Could the transition be extended just a little bit to perhaps .15 or .2 seconds? Especially since both text color and background color change, the rapid transition can be a bit extreme

Reply:
Sure, changed in https://github.com/opentripplanner/otp-ui/pull/470/commits/abd1d3fc03f9a6430464461e3c852aa127b9022b.